lib/puppet/pops/loader/puppet_function_instantiator.rb

```diff
@@ -64,18 +65,19 @@ def self.create(loader, typed_name, source_ref, pp_code_string)
   #   typed name, and an instantiated function with global scope closure associated with the given loader
   #
   def self.create_from_model(function_definition, loader)
-    closure_scope = Puppet.lookup(:global_scope) { {} }
-    created = create_function_class(function_definition, closure_scope)
+    closure_scope = nil; # Puppet.lookup(:global_scope) { {} }
+    created = create_function_class(function_definition)
     typed_name = TypedName.new(:function, function_definition.name)
     [typed_name, created.new(closure_scope, loader)]
   end
 
-  def self.create_function_class(function_definition, closure_scope)
+  def self.create_function_class(function_definition)
     # Create a 4x function wrapper around a named closure
     Puppet::Functions.create_function(function_definition.name, Puppet::Functions::PuppetFunction) do
+      # This is highly problematic - it binds both an Evaluator and closure_scope
```

Comment:
Half of it is obsolete - it still binds an evaluator - which it does not need to do - the evaluator holds on to a migration checker (it either creates a new null instance of one, or looks one up). If instead using a shared evaluator we would gain 2 object instances per function, and some performance. Changing the comment to a TODO wrt to the evaluator.new